db/seeds/schemas/add_agfs_fee_scheme_12.rb

```diff
@@ -0,0 +1,156 @@
+# Class to:
+# 1. create agfs fee scheme 12
+# 2. add an end date to fee scheme 11 (day before scheme 12 start)
+# 3. copy all fee scheme 11 offences, creating as fee scheme 12 records
+# 4. create offence fee scheme through table records to assoicate each
+#    with an fee scheme.
+#
+module Seeds
+  module Schemas
+    class AddAgfsFeeScheme12
+      attr_reader :pretend
+      alias_method :pretending?, :pretend
+
+      def initialize(pretend: false)
+        @pretend = pretend
+      end
+
+      def status
+         <<~STATUS
+          \sAGFS scheme 11 end date: #{agfs_fee_scheme_11&.end_date || 'nil'}
+          \sAGFS scheme 12 start date: #{agfs_fee_scheme_12&.start_date || 'nil'}
+          \sAGFS scheme 12 fee scheme: #{agfs_fee_scheme_12&.attributes || 'nil'}
+          \sAGFS scheme 12 offence count: #{scheme_12_offence_count}
+          \s------------------------------------------------------------
+          Status: #{agfs_fee_scheme_12.present? && scheme_12_offence_count > 0 ? 'up' : 'down'}
+          Enabled: #{Settings.agfs_scheme_12_enabled?}
+        STATUS
+      end
+
+      def up
+        modify_agfs_schemes_if_twelve_enabled
+      end
+
+      def down
+        destroy_agfs_scheme_12_offences
+        destroy_scheme_12_update_11
+      end
+
+      private
+
+      def agfs_fee_scheme_12
+        @agfs_fee_scheme_12 ||= FeeScheme.agfs.twelve.first
+      end
+
+      def agfs_fee_scheme_11
+        @agfs_fee_scheme_11 ||= FeeScheme.agfs.eleven.first
+      end
+
+      def destroy_agfs_scheme_12_offences
+        if pretending?
+          puts "Would delete #{scheme_12_offence_count} scheme 12 offences".yellow
+          puts "Would reset offence PK sequence to max id value".yellow
+        else
+          scheme_12_offences = Offence.joins(:fee_schemes).merge(FeeScheme.twelve).merge(FeeScheme.agfs).distinct
+          before_count = scheme_12_offences.count
+          puts "Deleted #{before_count} scheme 12 offences".green if scheme_12_offences.destroy_all
```

Comment:
Debug code?

Reply:
[ditto](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3432#discussion_r463064520)